### ISSUE TITLE
Picking objects in 3D has a delay.

### ISSUE THREAD

**Issue by viksl** [godotengine/godot#72621]
### Godot version

4.0 Beta 17 mono

### System information

Windows 10, Nvidia GTX 1660Ti, Vulkan

### Issue description

Check the video here: https://youtu.be/6xk_54SK0PU I tried to add mouse sounds too but my mic didn't pick them up for some reason.

You can see that picking objects has quite a delay in 3D, this is particularly highlighted when you pick objects in the node tree which is instantaneous (I udnerstand picking in 3D takes a bit more than processing the node tree, I'm just trying it to show somethign to contrast the delay with).

### Steps to reproduce

Open a 3d scene add some a MeshInstance, click on it, observe a delay before the gizmo appears after you click.

### Minimal reproduction project

-

**Comment by Calinou** [comment]
It's not an artificial delay (the editor is blocking during object picking). The object picking algorithm needs to be rewritten to be faster (possibly using the depth buffer and hardware readbacks).

**Comment by fkeyzuwu** [comment]
just chiming in to say this happens to me too, with AMD on 4.1 stable.

**Comment by GustJc** [comment]
Just a 'quick-fix' for anyone who might be having this.
I started noticing a horrible delay when selecting objects after updating to 4.1.2
Then I realized I downloaded the non-C# version of it when I updated.
I was previously using the .NET version, even though I'm only using gdscript.

After downloading the .NET version now, the select delay is more manageable.
I was getting always 2s+ delays when trying to select objects on Godot normal version.
On the c# version of Godot, it still have some minor delay, but it is not very noticeable. Probably less than 0.5s.

**tl;dr:**

- The delay is worst in the normal version of Godot (for me).
- Using the .NET Godot version improves the selection speed by a lot.


-------
On windows 10. (before 1903)
With a nvidea 1070, but last updated drivers from 1 year ago.

**Comment by nan0m** [comment]
I can confirm that this is also faster for me in the .net version. 

Godot v4.3.stable.mono - Windows 10.0.19045 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 3070 Laptop GPU (NVIDIA; 32.0.15.6636) - 11th Gen Intel(R) Core(TM) i7-11800H @ 2.30GHz (16 Threads)
